api/provider/aws/task_read.go

```diff
@@ -1,9 +1,90 @@
 package aws
 
 import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
 	"github.com/quintilesims/layer0/common/models"
 )
 
 func (t *TaskProvider) Read(taskID string) (*models.Task, error) {
-	return nil, nil
+	// todo: use tlake's
+	environmentID, err := t.lookupTaskEnvironmentID(taskID)
+	if err != nil {
+		return nil, err
+	}
+	fqEnvironmentID := addLayer0Prefix(t.Config.Instance(), environmentID)
+
+	taskARN, err := t.lookupTaskARN(taskID)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterName := fqEnvironmentID
```

Comment:
Again, this could be moved up with `fqEnvironmentID := ...` to avoid fragmenting logic.

Reply:
Again, ignore this comment.